cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -240,6 +286,47 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         return new ReconciliationState(reconciliation, kafkaAssembly);
     }
 
+    Future<Void> updateStatus(Reconciliation reconciliation, String namespace, String name, KafkaStatus desiredStatus) {
+        Future<Void> updateStatusFuture = Future.future();
+
+        crdOperator.getAsync(namespace, name).setHandler(getRes -> {
```

Comment:
I do not think this is how people expect it to work and it seems rather unintuitive. 

I guess I'm fine to use the original resource and just handle the 409 error and just not update it (although I do not necessarily see any reason for it).

But I do not think we should set the status to `NotReady` because of that. If you think that every new update is NotReady just because the resource changed we should have separate watch for this which would immediately after seeing any change invalidate the state and set it to `NotReady`. Doing this only at the end of the reconciliation will create a very strange behaviour when sometimes it is `NotReady` just because it was updated and sometimes it is still `Ready` for a long time after it was updates because for example the reconciliation is in progress and doing a rolling update. This will also make it hard to use from any tooling as the state will be changing constantly without indicating any issues.

I think this would be better addressed by including the version of the CR which was last reconciled than doing it this way.

Reply:
OK, yes, you're right we need `status.observedGeneration`.

Reply:
Arey ou fine to solve this in the next PR / next release?

Reply:
I guess so.

Reply:
Thank